Cabal/Distribution/Simple/Configure.hs

```diff
@@ -250,14 +260,18 @@ writePersistBuildConfig distPref lbi = do
   where
     pkgId = packageId $ localPkgDescr lbi
 
+-- | Identifier of the current Cabal package
 currentCabalId :: PackageIdentifier
 currentCabalId = PackageIdentifier (PackageName "Cabal") cabalVersion
 
+-- | Identifier of the current compiler package
 currentCompilerId :: PackageIdentifier
 currentCompilerId = PackageIdentifier (PackageName System.Info.compilerName)
                                       System.Info.compilerVersion
 
-parseHeader :: ByteString -> (PackageIdentifier, PackageIdentifier)
+-- | Parse the @setup-config@ file header, returning the package identifiers for Cabal and the compiler.
```

Comment:
https://github.com/haskell/cabal/wiki/Style-guide:
There has been some recent discussion about whether or not to adopt an official style guide. A consensus has not yet been reached,

So since there are no official guide lines, I commend your eagerness to refuse  extra documentation.